zetaclient/bitcoin_client_test.go

```diff
@@ -158,6 +175,63 @@ func (suite *BitcoinClientTestSuite) Test3() {
 	suite.Require().NoError(err)
 	suite.T().Logf("block number %d", bn)
 }
-func TestBitcoinChainClient(t *testing.T) {
-	suite.Run(t, new(BitcoinClientTestSuite))
+
+// func TestBitcoinChainClient(t *testing.T) {
+// 	suite.Run(t, new(BitcoinClientTestSuite))
+// }
+
+func TestBitcoinFeeRate(t *testing.T) {
+	// mainnet config
+	connCfg := &rpcclient.ConnConfig{
+		Host:         "127.0.0.1:8332",
+		User:         "username",
+		Pass:         "password",
+		Params:       "mainnet",
+		HTTPPostMode: true,
+		DisableTLS:   true,
+	}
```

Comment:
We can't use it for unit test
What about creating a mock for the client?
It doesn't seem we need anything more than a mock for `EstimateSmartFee`

Reply:
[commit](https://github.com/zeta-chain/node/pull/1713/commits/7ebaf3805a1dd7284917117e7e9b2f009c2fe037)

Reply:
Can we add a comment how this test is supposed to be run? We might consider adding this to the E2E test lists?

Reply:
I tried it on the Bitcoin testnet. The API returned me same number regardless of what `confTarget` `estimateMode` passed. What we expect in the test case is that the higher the `confTarget` passed, the lower the rate; and `Conservative` must give a higher gas rate than `Economical` if same `confTarget` is given. These two assertions will break in Testnet (I guess private network will also break the assertions).
![image](https://github.com/zeta-chain/node/assets/34498985/ccca37cc-f2db-4fde-a2db-c2a6cc551ac7)